### PR TITLE
Add logback config for scdf local server

### LINKED ISSUE #2554: Add logback appender file for rolling log file when using local server
The appender file can be parameterized to allow for some flexibility in configuration though properties.  See http://blog.wolfplusplus.com/?p=362

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Add logback configuration file
    - with both `console` and `rolling` file appender
    - Update rolling file appender with the defaults

 - Add `logging` configuration in application.yml

Resolves #2554

### CODE REVIEW

File: spring-cloud-dataflow-server-local/pom.xml [hunk 1]
```diff
@@ -85,6 +85,8 @@
 				<filtering>true</filtering>
 				<includes>
 					<include>application.yml</include>
+					<include>application.properties</include>
```
Comment (jvalkeal) [inline spring-cloud-dataflow-server-local/pom.xml]: We don't have `application.properties`?

**Comment by jvalkeal** [comment]
I really don't like the fact that this logs to `scdf-logs/spring-cloud-dataflow-server-local.log` on your current working directory regardless where the fat jar is located. I'd prefer log to go to `tmpdir` on default just like boot's default logback usually does it. It's ok to have a simple way to enable logging to a file if user wants it but I would not want these files on my system on default.

**Comment by ilayaperumalg** [comment]
ok, Boot has this:

```
<property name="LOG_FILE" value="${LOG_FILE:-${LOG_PATH:-${LOG_TEMP:-${java.io.tmpdir:-/tmp}}}/spring.log}"/>
```
and we can probably use the same with the change in the log file name.

**Comment by jvalkeal** [comment]
Yeah, that's what I meant. Just use different default log file name.

File: spring-cloud-dataflow-server-local/src/main/resources/logback-scdf-local.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+	<conversionRule conversionWord="clr" converterClass="org.springframework.boot.logging.logback.ColorConverter" />
+	<conversionRule conversionWord="wex" converterClass="org.springframework.boot.logging.logback.WhitespaceThrowableProxyConverter" />
+	<conversionRule conversionWord="wEx" converterClass="org.springframework.boot.logging.logback.ExtendedWhitespaceThrowableProxyConverter" />
+	<property name="CONSOLE_LOG_PATTERN" value="${CONSOLE_LOG_PATTERN:-%clr(%d{${LOG_DATEFORMAT_PATTERN:-yyyy-MM-dd HH:mm:ss.SSS}}){faint} %clr(${LOG_LEVEL_PATTERN:-%5p}) %clr(${PID:- }){magenta} %clr(---){faint} %clr([%15.15t]){faint} %clr(%-40.40logger{39}){cyan} %clr(:){faint} %m%n${LOG_EXCEPTION_CONVERSION_WORD:-%wEx}}"/>
+	<property name="FILE_LOG_PATTERN" value="${FILE_LOG_PATTERN:-%d{${LOG_DATEFORMAT_PATTERN:-yyyy-MM-dd HH:mm:ss.SSS}} ${LOG_LEVEL_PATTERN:-%5p} ${PID:- } --- [%t] %-40.40logger{39} : %m%n${LOG_EXCEPTION_CONVERSION_WORD:-%wEx}}"/>
+
+	<logger name="org.springframework.beans" level="WARN"/>
```
Comment (markpollack) [inline spring-cloud-dataflow-server-local/src/main/resources/logback-scdf-local.xml]: Is this still going to be possible to override via env-vars?

**Comment by markpollack** [comment]
rebase, squashed, merged as f76ca18694f8f7dd9747cda6803e32464693fe3a

**Pull request closed without merging**